### PR TITLE
Cli load nipt

### PR DESCRIPTION
**Pull request by mayabrandi**
## CLI upload to statina
*This PR adds cli for loading into nipt database*

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh cli_load_nipt`
` bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-servers-stage.sh niptool_internal`

### How to test
- [x] ` cg upload nipt statina case tendercora`

## Review
- [x] code approved by MR
- [x] tests executed by MB
- [x] "Merge and deploy" approved by MR
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/upload/nipt/ftp.py [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
     pass
 
 
-@nipt.command("case")
+@ftp.command("case")
 @click.argument("case_id", required=True)
 @click.option("--dry-run", is_flag=True)
 @click.pass_obj
-def nipt_upload_case(context: CGConfig, case_id: str, dry_run: bool):
+def nipt_upload_case(configs: CGConfig, case_id: str, dry_run: bool):
```
Comment (Mropat) [inline cg/cli/upload/nipt/ftp.py:23]: can we still call it context for consistency with rest of cg code?

File: cg/meta/upload/nipt/models.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from pathlib import Path
+from typing import Optional
+from pydantic import BaseModel
+
+
+class UpploadFiles(BaseModel):
+    result_file: Path
+    multiqc_report: Optional[Path]
+    segmental_calls: Optional[Path]
```
Comment (Mropat) [inline cg/meta/upload/nipt/models.py:9]: segmental calls will be in multiple files
like so:
```
2021-05-27 15:26:59 hasta.scilifelab.se housekeeper.store.api.find[62294] INFO Fetching files from version 40011
                                                                        📜 Files table 📜                                                                        
┏━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ ID     ┃ File name                                                                                                     ┃ Tags                                 ┃
┡━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ 952006 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2105187.csv                                   │ metrics, output, nipt                │
│ 952007 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/multiqc_report.html                           │ multiqc-html, output, nipt           │
│ 952008 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2020-01862-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2020-01862-05 │
│ 952009 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2020-01929-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2020-01929-05 │
│ 952010 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2020-17754-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2020-17754-05 │
│ 952011 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2020-31120-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2020-31120-05 │
│ 952012 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-06644-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-06644-05 │
│ 952013 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-10263-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-10263-05 │
│ 952014 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-11631-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-11631-05 │
│ 952015 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12713-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12713-05 │
│ 952016 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12714-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12714-05 │
│ 952017 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12715-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12715-05 │
│ 952018 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12716-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12716-05 │
│ 952019 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12717-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12717-05 │
│ 952020 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12718-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12718-05 │
│ 952021 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12759-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12759-05 │
│ 952022 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12761-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12761-05 │
│ 952023 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12763-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12763-05 │
│ 952024 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12764-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12764-05 │
│ 952025 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12765-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12765-05 │
│ 952026 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12768-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12768-05 │
│ 952027 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12792-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12792-05 │
│ 952028 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12794-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12794-05 │
│ 952029 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12828-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12828-05 │
│ 952030 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12829-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12829-05 │
│ 952031 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12862-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12862-05 │
│ 952032 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12864-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12864-05 │
│ 952033 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12865-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12865-05 │
│ 952034 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12866-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12866-05 │
│ 952035 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12867-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12867-05 │
│ 952036 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12868-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12868-05 │
│ 952037 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12937-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12937-05 │
│ 952038 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12938-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12938-05 │
│ 952039 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12939-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12939-05 │
│ 952040 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12940-04.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12940-04 │
│ 952041 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12941-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12941-05 │
│ 952042 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12966-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12966-05 │
│ 952043 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-12967-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-12967-05 │
│ 952044 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13036-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13036-05 │
│ 952045 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13037-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13037-05 │
│ 952046 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13038-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13038-05 │
│ 952047 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13039-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13039-05 │
│ 952048 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13040-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13040-05 │
│ 952049 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13041-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13041-05 │
│ 952050 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13042-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13042-05 │
│ 952051 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13043-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13043-05 │
│ 952052 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13044-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13044-05 │
│ 952053 │ /home/proj/production/housekeeper-bundles/civilskunk/2021-05-20/2021-13045-05.WCXpredict_aberrations.filt.bed │ cnv, nipt, wisecondor, 2021-13045-05 │
└────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────┘

```

File: cg/meta/upload/nipt/nipt.py [hunk 3]
```diff
@@ -105,3 +107,32 @@ def update_analysis_upload_started_date(self, case_id: str) -> models.Analysis:
...
+            "POST",
+            f"{self.niptool_host}/insert/batch",
+            "-H",
+            "accept: application/json",
+            "-H",
+            "Content-Type: application/json",
+            "-d",
+            str(upload_files.dict(exclude_none=True)),
+        ]
+        self.process.run_command(parameters=load_call)
```
Comment (Mropat) [inline cg/meta/upload/nipt/nipt.py:134]: let's use requests library for this"

File: cg/cli/upload/nipt/niptool.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        hk_segmental_calls_file: str = nipt_upload_api.get_housekeeper_results_file(
+            case_id=case_id, tags=["nipt", "wisecondor"]
+        )
+        segmental_calls_file: Path = nipt_upload_api.get_results_file_path(hk_segmental_calls_file)
+
+        nipt_upload_api.upload_to_niptool_database(
+            results_file=results_file,
+            multiqc_file=multiqc_file,
+            segmental_calls_file=segmental_calls_file.parents[0],
```
Comment (Mropat) [inline cg/cli/upload/nipt/niptool.py:48]:
```suggestion
            segmental_calls_file=segmental_calls_file.parent.as_posix(),
```

**Comment by mayabrandi** [comment]
add tests maya!!!!!! @mayabrandi

**Comment by mayabrandi** [comment]
Testing acording to above..

Skärmavbild 2021-05-31 kl  14 29 41
👍

File: CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,7 @@ __________ DO NOT TOUCH ___________
 
 ## [x.x.x]
 ### Added
+- cli for upload of data to statina
```
Comment (Mropat) [inline CHANGELOG.md:17]: please take a look at the changelog file header

File: cg/cli/upload/nipt/statina.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+        hk_multiqc_file: str = nipt_upload_api.get_housekeeper_results_file(
+            case_id=case_id, tags=["nipt", "multiqc-html"]
+        )
+        multiqc_file: Path = nipt_upload_api.get_results_file_path(hk_multiqc_file)
+
+        hk_segmental_calls_file: str = nipt_upload_api.get_housekeeper_results_file(
+            case_id=case_id, tags=["nipt", "wisecondor"]
+        )
+        segmental_calls_file: Path = nipt_upload_api.get_results_file_path(hk_segmental_calls_file)
```
Comment (Mropat) [inline cg/cli/upload/nipt/statina.py:43]: I suggest moving this into the upload_to_statina method so all the file fetching happens in the API

**Review by Mropat — APPROVED** [review]
:fire:  AMAZING work!! :fire: 

Please update changelog version and ready to merge!

Is there anyting that needs to be deployed in servers at the same time?

**Pull request merged**